KalturaWowzaServer/src/main/java/com/kaltura/media_server/services/KalturaEntryDataPersistence.java

```diff
@@ -23,49 +23,58 @@ public static void setAppInstance(IApplicationInstance appInstance) {
+            logger.debug("Persistence hash map cleanup will start in " + Constants.KALTURA_ENTRY_PERSISTENCE_CLEANUP_START / 1000 + " seconds");
+        }
+        else {
+            logger.debug("Persistence cleanup was called less than " + Constants.KALTURA_TIME_BETWEEN_PERSISTENCE_CLEANUP / 1000 + " seconds ago. Ignoring call!");
+        }
 	}
 
 	private static void CleanUp() {
-        synchronized (entryIdToKalturaLiveEntryMap) {
-            // Do not perform entries cleanup more than once a minute, to prevent a load of timer tasks running at the same time.
-            long currentTime = System.currentTimeMillis();
-            if (currentTime - lastMapCleanUp > Constants.KALTURA_TIME_BETWEEN_PERSISTENCE_CLEANUP) {
-                try {
-                    Set<String> playingEntriesList = Utils.getEntriesFromApplication(_appInstance);
-                    Set<String> hashedEntriesList = entryIdToKalturaLiveEntryMap.keySet();
-                    long currTime = System.currentTimeMillis();
-                    for (String entry : hashedEntriesList) {
-                        // Check start time to avoid a race between the thread adding the entry to the map and this
-                        // current thread that wants to remove it. Worst case scenario entry will ve erased in the next run.
-                        long startTime = (long) getPropertyByEntry(entry, Constants.KALTURA_ENTRY_START_TIME);
-                        if ((!playingEntriesList.contains(entry)) && (currTime - startTime > Constants.KALTURA_PERSISTENCE_DATA_MIN_ENTRY_TIME)) {
-                            logger.info("Entry " + entry + " no longer exists. Removing it from Persistence Hash Map");
+	    try {
+            synchronized (entryIdToKalturaLiveEntryMap) {
+                Set<String> playingEntriesList = Utils.getEntriesFromApplication(_appInstance);
+                Set<String> hashedEntriesList = entryIdToKalturaLiveEntryMap.keySet();
+                long currentTime = System.currentTimeMillis();
+                for (String entry : hashedEntriesList) {
+                    // Check start time to avoid a race between the thread adding the entry to the map and this
+                    // current thread that wants to remove it. Worst case scenario entry will ve erased in the next run.
+                    long startTime = (long)getPropertyByEntry(entry, Constants.KALTURA_ENTRY_START_TIME);
+                    if (!playingEntriesList.contains(entry)) {
+                        logger.info("(" + entry + ") Entry no longer exists");
+                        int minTimeInMap = Constants.KALTURA_PERSISTENCE_DATA_MIN_ENTRY_TIME;
+                        if (currentTime - startTime > minTimeInMap) {
+                            logger.info("(" + entry + ") Entry start time is greater than " + minTimeInMap / 1000 + " seconds; Removing it from Map!");
                             entryIdToKalturaLiveEntryMap.remove(entry);
```

Comment:
missing log message for the else {logger.debug("(" + entry + ") Entry start time is less than " + minTimeInMap / 1000 + " seconds;  Not removing it from Map though stream is not live!");}